### DISCUSSION TITLE
'Dns01Precondition' failed: "The delegated name server is not correct. DNS zone = narwolves.net, Expected = ns1-02.azure-dns.com,ns2-02.azure-dns.net,ns3-02.azure-dns.org,ns4-02.azure-dns.info, Actual = "

### DISCUSSION THREAD

**Discussion by kicka5h** [polymind-inc/acmebot#485] [Q&A] [answered]
Looks like it's not returning the proper value for name servers. I have verified that nslookup is returning the correct expected values for NS and SOA. How can I resolve this?

![image](https://user-images.githubusercontent.com/35403561/169626461-ea193c9f-dc93-46c7-9acf-165f67a6229b.png)

<img width="303" alt="DNS" src="https://user-images.githubusercontent.com/35403561/169626419-ce045596-e15a-4c53-b533-44c559449cb8.png">

**Comment by shibayan** [comment, marked as answer]
Acmebot uses Google Public DNS to check DNS settings, so query the DNS server 8.8.8.8. Your domain was not reflected in my environment. It may take up to 48 hours.

![image](https://user-images.githubusercontent.com/1356444/169628606-2dce9045-674c-4d75-8393-b14f93f0d9df.png)

**Reply by shibayan** [reply]
This seems to be a problem not with Acmebot but with incorrectly configured DNSSEC records.
Please review your settings to ensure correct DNS record resolution.

**Comment by kicka5h** [comment]
It's been over a week since I delegated this domain to Azure. Looks like it hasn't fully propagated yet I guess though.. if you check https://www.whatsmydns.net/#NS/narwolves.net, there are still quite a few places not reporting the name servers.

**Comment by georgegil** [comment]
got a similiar issue here. 

we use custom nameserver names in Cloudlfare, and acmebot is not liking this:

`Orchestrator function 'IssueCertificate' failed: The activity function 'Dns01Precondition' failed: "The delegated name server is not correct. DNS zone = vandealer2dealer.com, Expected = cullen.ns.cloudflare.com,deborah.ns.cloudflare.com, Actual = ns1.vandealer2dealer.com,ns2.vandealer2dealer.com".`

**Reply by shibayan** [reply]
If your Cloudflare zone includes custom Name servers in the `vanity_name_servers`, you can easily handle this.

I don't have the environment to verify custom Name servers, though.

**Reply by georgegil** [reply]
I am not quite following sorry. 🥺

But happy to try what you are suggesting, can you perhaps give me some guidance?

**Reply by shibayan** [reply]
If you follow the steps in this document, I would be able to attempt to fix it at the code level. This will not affect existing Cloudflare users.
https://developers.cloudflare.com/dns/additional-options/custom-nameservers/

**Reply by georgegil** [reply]
yes we have custom (vanity) name servers enabled. 

I am not allowed to disable this alas.

**Reply by shibayan** [reply]
I have attempted to fix this in the latest version. I hope this fix fixes the problem.
https://github.com/shibayan/keyvault-acmebot/releases/tag/v4.0.0-preview2

**Reply by georgegil** [reply]
That worked!! 👍🙏
